WalletWasabi/Wallets/Wallet.cs

```diff
@@ -257,6 +265,36 @@ public override async Task StartAsync(CancellationToken cancel)
 		}
 	}
 
+	private async Task PerformFinalSynchronizationAsync(CancellationToken cancel)
+	{
+		using (await HandleFiltersLock.LockAsync(cancel).ConfigureAwait(false))
+		{
+			while (!cancel.IsCancellationRequested)
+			{
+				try
+				{
+					await PerformWalletSynchronizationAsync(SyncType.NonTurbo, cancel).ConfigureAwait(false);
+					if (LastProcessedFilter is { } lastProcessedFilter)
+					{
+						SetFinalBestHeight(new Height(lastProcessedFilter.Header.Height));
+					}
+
+					break;
+				}
+				catch (InvalidOperationException ex)
+				{
+					// Retry until cancellation is requested
+					Logger.LogWarning($"Final synchronization encountered an error while processing filter {LastProcessedFilter?.Header.Height}, retrying: '{ex}'.");
```

Comment:
Retrying forever also means filling the log file with endless amount of errors.

The question here is: What does the `InvalidOperationException` means here? Can we do something about it? If there is not thing than can be done then the best is to do nothing. Remember: the first rule of exception handling is DO NOT catch exceptions.